### PR TITLE
Exclude grpc package from pulsar-broker

### PR DESCRIPTION
**Pull request by gaoran10**
### Motivation

Currently, the pulsar-broker has a conflict dependency problem. This will cause failure when building the project. And we need to resolve the dependency problem in Pulsar, after Pulsar fixes the problem, we could improve this change.

```
Cannot resolve Could not resolve version conflict among

io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> org.apache.bookkeeper:stream-storage-server:jar:4.14.4 -> org.apache.bookkeeper:stream-storage-java-client:jar:4.14.4 -> io.grpc:grpc-core:jar:1.42.1
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> org.apache.bookkeeper:stream-storage-server:jar:4.14.4 -> org.apache.bookkeeper:stream-storage-java-client:jar:4.14.4 -> io.grpc:grpc-testing:jar:1.42.1 -> io.grpc:grpc-core:jar:[1.42.1,1.42.1]
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> io.grpc:grpc-all:jar:1.45.1 -> io.grpc:grpc-core:jar:[1.45.1,1.45.1]
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> io.grpc:grpc-all:jar:1.45.1 -> io.grpc:grpc-grpclb:jar:1.45.1 -> io.grpc:grpc-core:jar:[1.45.1,1.45.1]
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> io.grpc:grpc-all:jar:1.45.1 -> io.grpc:grpc-netty:jar:1.45.1 -> io.grpc:grpc-core:jar:[1.45.1,1.45.1]
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> io.grpc:grpc-all:jar:1.45.1 -> io.grpc:grpc-rls:jar:1.45.1 -> io.grpc:grpc-core:jar:[1.45.1,1.45.1]
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> io.grpc:grpc-all:jar:1.45.1 -> io.grpc:grpc-services:jar:1.45.1 -> io.grpc:grpc-core:jar:[1.45.1,1.45.1]
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> io.grpc:grpc-all:jar:1.45.1 -> io.grpc:grpc-xds:jar:1.45.1 -> io.grpc:grpc-core:jar:1.45.1
io.streamnative:pulsar-broker:jar:2.9.2.14 -> io.streamnative:pulsar-zookeeper-utils:jar:2.9.2.14 -> io.grpc:grpc-all:jar:1.45.1 -> io.grpc:grpc-xds:jar:1.45.1 -> io.grpc:grpc-netty-shaded:jar:[1.45.1,1.45.1] -> io.grpc:grpc-core:jar:[1.45.1,1.45.1]
```

### Modifications

Exclude the grpc package from pulsar-broker.

### CODE REVIEW

**Review by Demogorgon314 — APPROVED** [review]

File: kafka-impl/pom.xml [hunk 1]
```diff
@@ -87,6 +87,12 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <scope>test</scope>
+    </dependency>
+
```
Comment (BewareMyPower) [inline kafka-impl/pom.xml:95]: Could you explain why do you add the JUnit dependency?

**Comment by BewareMyPower** [comment]
> Currently, the pulsar-broker has a conflict dependency problem.

Is there any related issue in Pulsar?

File: kafka-impl/pom.xml (hunk 1 above)
Comment (Demogorgon314) [reply kafka-impl/pom.xml:95]: See `DelayedOperationTest`, this test used JUnit, I'll push a PR to fix it.

Comment (gaoran10) [reply kafka-impl/pom.xml:95]: Yes, before excluding, the JUnit comes from `grpc-testing` package.

**Comment by gaoran10** [comment]
Hi, @BewareMyPower , I create an issue https://github.com/apache/pulsar/issues/15290 in apache Pulsar. PTAL

**Comment by Demogorgon314** [comment]
@gaoran10 You need to sign the license agreement to pass the CI check : )

**Pull request merged**